### ISSUE TITLE
How to view the ffmpeg command line generated by fluent ffmpeg?

### ISSUE THREAD

**Issue by uniquegodwin** [fluent-ffmpeg/node-fluent-ffmpeg#716]
Hi,
How do I get the generated command that's being executed by fluent ffmpeg?
Is there a function that would return this string?

Thanks.

**Comment by coyotte508** [comment]
+1

I'd like to know what the error message of ffmpeg and having the command line would help.

**Comment by njoyard** [comment]
Guys, please read the docs first...
start event with command line : https://github.com/fluent-ffmpeg/node-fluent-ffmpeg#start-ffmpeg-process-started
ffmpeg console output : https://github.com/fluent-ffmpeg/node-fluent-ffmpeg#stderr-ffmpeg-output

**Comment by forresthopkinsa** [comment]
@njoyard Is there a way to get the command line *without* spawning an ffmpeg process? I'm wondering if this tool can be used just to create the command string.

**Comment by gclsoft** [comment]
how to print?

**Comment by letoribo** [comment]
`.on('start', (cmdline) => console.log(cmdline))`

[https://github.com/fluent-ffmpeg/node-fluent-ffmpeg#start-ffmpeg-process-started](url)

**Comment by gclsoft** [comment]
```
  createCommand(conf = {}) {
    const {threads = 1} = conf;
    const command = ffmpeg();
    if (threads > 1) command.addOptions([`-threads ${threads}`]);
    console.log("###:"+command)
    return command;
  },

    this.command = FFmpegUtil.createCommand();
    this.command.on('start', (cmdline) => {
      console.log(cmdline)
    })
```
@letoribo  Not work. How to print it?

**Comment by gclsoft** [comment]
```
    const command = ffmpeg();
    command.on('start', (cmdline) => {
      console.log(cmdline)
    })
```
@letoribo Not work. How to print it?

**Comment by gclsoft** [comment]
It does work. Thanks!

**Comment by soyniqolas** [comment]
TLDR: 

```
ffmpeg('/path/to/file.avi')
  .on('stderr', function(stderrLine) {
    console.log('Stderr output: ' + stderrLine);
  });
```

**Comment by sixman9** [comment]
I'm new to this project, but would also like the command [line] output. 

Can I assume that the installed FFMPEG instance will begin to process its file(s) on 'start'? If so, **is there any way of logging the command line WITHOUT executing an FFMPEG session?**

I need to pass the CLi to an external FFMPEG instance, so I do not need anything actually process.

Basically, I like your API, but want to possibly use it on mobile (no FFMPEG PATH available).

**Comment by sixman9** [comment]
> ```
>     const command = ffmpeg();
>     command.on('start', (cmdline) => {
>       console.log(cmdline)
>     })
> ```
> 
> @letoribo Not work. How to print it?

*Does this require `.run()` to be executed for the 'start' event closure to be executed* (I will test this as soon as I put together a test project)?

Put another way, does the FFMPEG job have to actually be executed, in order to get the command-line output?

I need to be able to get the output *without* actually processing any files.

**Comment by serg06** [comment]
@sixman9 After looking through the source code, I see two options:

1. `FfmpegCommand._getArguments()` [here](https://github.com/fluent-ffmpeg/node-fluent-ffmpeg/blob/68d5c948b689b3058e52435e0bc3d4af0eee349e/lib/processor.js#L223), which figures out the args passed to ffmpeg. (It's exposed in `@types/fluent-ffmpeg`.)
2. `FfmpegCommand._prepare()` [here](https://github.com/fluent-ffmpeg/node-fluent-ffmpeg/blob/68d5c948b689b3058e52435e0bc3d4af0eee349e/lib/processor.js#L290), which does the same thing, but does a does a bunch of work to determine if it needs to append `-strict experimental` to the args before passing them to you. (It's not exposed in `@types/fluent-ffmpeg`.)

**Comment by sixman9** [comment]
> ... I see two options:
> 
> 1. `FfmpegCommand._getArguments()` [here](https://github.com/fluent-ffmpeg/node-fluent-ffmpeg/blob/68d5c948b689b3058e52435e0bc3d4af0eee349e/lib/processor.js#L223), which figures out the args passed to ffmpeg. (It's exposed in `@types/fluent-ffmpeg`.)
> 2. `FfmpegCommand._prepare()` [here](https://github.com/fluent-ffmpeg/node-fluent-ffmpeg/blob/68d5c948b689b3058e52435e0bc3d4af0eee349e/lib/processor.js#L290), which does the same thing, but does a does a bunch of work to determine if it needs to append `-strict experimental` to the args before passing them to you. (It's not exposed in `@types/fluent-ffmpeg`.)

#2 seems to involve the use off FFProbe. To expand on what I was trying to do, I'd like to use the domain knowledge (arguments) from this project to drive a client-side-only [FFMpegWASM](https://github.com/ffmpegwasm/ffmpeg.wasm) invocation, so, browser-based FFMpeg usage, without a PATH or file system (in-browser).

#1 might help (arguments with no attempt to call FFMpeg or FFProbe).

I don't know how important `-strict experimental` but I'll guess I'll find out.
